http/custom_roundtripper.go

```diff
@@ -0,0 +1,55 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	awsAuth "github.com/ONSdigital/dp-elasticsearch/v2/awsauth"
+)
+
+type awsSignerRoundTripper struct {
+	signer       *awsAuth.Signer
+	roundTripper http.RoundTripper
+}
+
+func NewAWSSignerRoundTripper(awsFilename, awsProfile, awsRegion, awsService string, customTransport http.RoundTripper) http.RoundTripper {
+	var roundTripper http.RoundTripper
+	if awsRegion == "" || awsService == "" {
+		log.Fatal(context.Background(), "aws region and service should be valid options")
```

Comment:
Should you not be returning an error here?

Reply:
This is a standard initialiser pattern in golang. The log fatal will log the error and return with an exit code of 1.

Reply:
for example, this follows the pattern used: https://github.com/up9inc/mizu/blob/main/cli/cmd/root.go

```
var rootCmd = &cobra.Command{
	Use:   "mizu",
	Short: "A web traffic viewer for kubernetes",
	Long: `A web traffic viewer for kubernetes
Further info is available at https://github.com/up9inc/mizu`,
	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
		if err := config.InitConfig(cmd); err != nil {
			logger.Log.Fatal(err)
		}

		return nil
	},
}
```

Reply:
commented below